### PR TITLE
Make need for add'l commit after adding .dockstore.yml explicit + clearer

### PR DESCRIPTION
**Pull request by aofarrel**
Every time you create a new .dockstore.yml file to add a new repo to Dockstore, you must make one additional commit to the repo *after* the .dockstore.yml is committed, or else your wf/tool/service will not show up. Despite what some of our older docs implied, this is the case even if you previously gave the GitHub app access to all repositories and/or already have a GitHub app wf/tool/service on Dockstore for that same organization/user.

### CODE REVIEW

**Review by coverbeck — COMMENTED** [review]
> Every time you create a new .dockstore.yml file to add a new repo to Dockstore, you must make one additional commit to the repo after the .dockstore.yml is committed, or else your wf/tool/service will not show up.

I don't think this is correct.

1. Enable GitHub Repo for repo foo
2. Add/commit valid .dockstore.yml to repo foo locally on branch goo
3. Push  branch goo
4. The workflow should show up in Dockstore.

Are you saying if you do that, the workflow doesn't appear? If it doesn't, that sounds like a bug.

Also, if you target the develop branch with the PR, this will appear only when we release the 1.13 doc, not before. Given that you have started your branch with `hotfix`, I'm guessing that's not your intent.

File: docs/getting-started/github-apps/note--addl-commit.rst [hunk 1]
```diff
@@ -0,0 +1,2 @@
+.. note::
+	Once the GitHub App is installed and a .dockstore.yml is present, please make sure to push one *additional* commit to your repository *after* the commit that adds the .dockstore.yml file. This helps make sure your workflows, tools, and services show up in Dockstore.
```
Comment (denis-yuen) [inline docs/getting-started/github-apps/note--addl-commit.rst:2]: FWIW, I think this is a little unclear. We need a commit of some kind after adding the repository in the github apps dialogue on github. That can either be "real" code or just adding/modifying the .dockstore.yml but there's nothing special about the .dockstore.yml that would disallow it from being a commit that would trigger the update process.

**Review by denis-yuen — CHANGES_REQUESTED** [review]
or what Charles said

**Comment by aofarrel** [comment]
> > Every time you create a new .dockstore.yml file to add a new repo to Dockstore, you must make one additional commit to the repo after the .dockstore.yml is committed, or else your wf/tool/service will not show up.
> 
> I don't think this is correct.
> 
> 1. Enable GitHub Repo for repo foo
> 2. Add/commit valid .dockstore.yml to repo foo locally on branch goo
> 3. Push  branch goo
> 4. The workflow should show up in Dockstore.
> 
> Are you saying if you do that, the workflow doesn't appear? If it doesn't, that sounds like a bug.
> 
> Also, if you target the develop branch with the PR, this will appear only when we release the 1.13 doc, not before. Given that you have started your branch with `hotfix`, I'm guessing that's not your intent.

My understanding is that you would need to push one **additional** commit to branch goo before anything showed up on Dockstore, based on an experience I had last week:

1. (months ago) Install docktore github app on aofarrel so that it has access to **all** repositories
2. (months ago) Add a few aofarrel repos to Dockstore (covstats, ORFalyzer, etc) via the Dockstore GitHub App
3. (last week) Fork someone else's repo of workflows onto aofarrel (let's call it forkfoo)
4. Create a valid .dockstore.yml for forkfoo locally
5. Push that valid .dockstore.yml for forkfoo's four workflows
6. Notice that none of forkfoo's workflows are showing up in Dockstore's unpublished tab
7. Notice that forkfoo *does* show up in the Dockstore GitHub app logs as "success" and "install" at 12:33 PM
8. Be puzzled; decide to go do something else for three hours
9. Shrug, add a space to forkfoo's readme, and push it
10. Forkfoo's four repos suddenly appear in Dockstore's unpublished tab
11. Notice that forkfoo *now* shows up in the Dockstore GitHub app logs with a new entry as "success" and "push" at 15:35 PM

If this is a bug I can report it as such, but until that bug is fixed we should document how to work around it (ie make an additional push on every new repo after the .dockstore.yml is pushed)

File: docs/getting-started/github-apps/note--addl-commit.rst (hunk 1 above)
Comment (aofarrel) [reply docs/getting-started/github-apps/note--addl-commit.rst:2]: In my experience the addition of .dockstore.yml triggers a successful install, but it is not until a successful push that it appears on Dockstore (see comment below).

**Comment by aofarrel** [comment]
Currently trying to replicate on a brand-new repo, just to make sure it's not a fluke.

**Comment by aofarrel** [comment]
@denis-yuen @coverbeck Made a ticket documenting what I'm seeing in two different scenarios. --> https://github.com/dockstore/dockstore/issues/4831

Unless the issue is patched quickly or it is blatant user error on my part, I think this PR deserves to be in our docs. Bug or not, it seems to have the potential to be tripping people up.

**Comment by coverbeck** [comment]
> @denis-yuen @coverbeck Made a ticket documenting what I'm seeing in two different scenarios. --> [dockstore/dockstore#4831](https://github.com/dockstore/dockstore/issues/4831)
> 
> Unless the issue is patched quickly or it is blatant user error on my part, I think this PR deserves to be in our docs. Bug or not, it seems to have the potential to be tripping people up.

I commented on that ticket, and I'm still not convinced there's an issue. I think it adds confusion saying you need to push twice for your changes to show up. And would you have to do it every time you change the workflow, or just the first time you push, i.e., do other updates have the same issue?

If there is a bug we should fix it ASAP, but we need a reproducible and/or researchable case.

If you can give us an approximate day/time when an update didn't go through, let us know the workflow and timestamp, and we can take a look at the logs.

**Comment by aofarrel** [comment]
> > @denis-yuen @coverbeck Made a ticket documenting what I'm seeing in two different scenarios. --> [dockstore/dockstore#4831](https://github.com/dockstore/dockstore/issues/4831)
> > Unless the issue is patched quickly or it is blatant user error on my part, I think this PR deserves to be in our docs. Bug or not, it seems to have the potential to be tripping people up.
> 
> I commented on that ticket, and I'm still not convinced there's an issue. I think it adds confusion saying you need to push twice for your changes to show up. And would you have to do it every time you change the workflow, or just the first time you push, i.e., do other updates have the same issue?
> 
> If there is a bug we should fix it ASAP, but we need a reproducible and/or researchable case.
> 
> If you can give us an approximate day/time when an update didn't go through, let us know the workflow and timestamp, and we can take a look at the logs.

It was March 31st. Can't recall exact times but I screenshot the logs in the ticket; 12:33 registered the install, while it was not until 15:35 after a new push that the apps register a push event. Unfortunately I can't link you to a specific commit as I did some squashing to clean up the PR I was making against the original fragpipe repo, but this is the repo: https://github.com/aofarrel/fragpipe-proteomics-pipeline-tutorial

Basically I was making sure my .dockstore.yml was valid before opening a PR for Dave.

**Comment by aofarrel** [comment]
Originally posted here but I'll move the debugging aspect to the ticket itself for clarity.

File: docs/getting-started/github-apps/note--addl-commit.rst (hunk 1 above)
Comment (aofarrel) [reply docs/getting-started/github-apps/note--addl-commit.rst:2]: I interpreted this existing FAQ entry...
> If access was given to the wrong organization or repository, or this is your first time installing the app, you’ll need to push another commit after adding the correct repository to activate the sync to Dockstore.

...to mean that an additional commit is needed after any fresh install. But to clarify, you're saying that the fresh-install-needs-extra-commit thing applies iff a .dockstore.yml was present prior to the installation?

Comment (aofarrel) [reply docs/getting-started/github-apps/note--addl-commit.rst:2]: @denis-yuen Seeking clarification before I close/edit this PR ^

**Comment by aofarrel** [comment]
Closing this -- https://github.com/dockstore/dockstore/issues/4856, which targets 1.12.x, should address this issue. Unless https://github.com/dockstore/dockstore/issues/4856 gets pushed back, this note in the docs is probably not needed.
